### PR TITLE
Resolve QPCleaner issues

### LINKED ISSUE #185: ReadEnvelope fails for large string in a single line.
What I did:
Run TestParseMimeTree with the file
https://www.mediafire.com/file/27w6hosijzd85pi/Get_started_with_OneDrive.eml/file
.

What I expected:
no error and HTML, text, attachments of email.

What I got:
envelope_test.go:87: Failed to parse MIME: Failed to ReadParts: multiple Read calls return no data or error

Release or branch I am using:
master branch. commit no: 6b038639b93ca41ddebf246745595190bd22d491

(Please attach a sample message if you feel it will help reproduce the issue)
Check parseParts method. The buffer is getting full. If we shorten or delete line#152 of Get_started_with_OneDrive.eml. Everything works.

Thanks

### PR DESCRIPTION
**Pull request by jhillyerd**
- QPCleaner: support tiny destination buffers
- QPCleaner: break long lines

Fixes #185

### CODE REVIEW

**Comment by jhillyerd** [comment]
Benchmark against master b3ffb18

```
goos: linux
goarch: amd64
pkg: github.com/jhillyerd/enmime/internal/coding
cpu: AMD Ryzen 7 2700 Eight-Core Processor
BenchmarkQPCleaner-16            1854934               628.3 ns/op      59048198.31 MB/s
PASS
ok      github.com/jhillyerd/enmime/internal/coding     1.848s
```

Benchmark against this PR

```
goos: linux
goarch: amd64
pkg: github.com/jhillyerd/enmime/internal/coding
cpu: AMD Ryzen 7 2700 Eight-Core Processor
BenchmarkQPCleaner-16            2215791               543.2 ns/op      81575889.62 MB/s
PASS
ok      github.com/jhillyerd/enmime/internal/coding     1.778s
```

🏇 💨

**Comment by jhillyerd** [comment]
PTAL if you have time in the next few days @requaos or @dcormier   Cheers.

File: internal/coding/quotedprint.go [hunk 1]
```diff
@@ -9,33 +9,87 @@ import (
...
+		if qp.lineLen >= MaxQPLineLen {
+			writeBytes(lineBreak)
+			qp.lineLen = 0
+		}
+
 		switch {
+		// Pass valid hex bytes through, otherwise escapes the equals symbol.
 		case b == '=':
-			// Pass valid hex bytes through.
+			ensureLineLen(2)
```
Comment (dcormier) [inline internal/coding/quotedprint.go:91]: I could be misreading, but shouldn't these calls be requesting `3` bytes?
```suggestion
			ensureLineLen(3)
```

File: internal/coding/quotedprint.go [hunk 2]
```diff
@@ -45,30 +99,38 @@ func (qp *QPCleaner) Read(dest []byte) (n int, err error) {
...
+		case b == '\r' || b == '\n':
+			writeByte(b)
+			qp.lineLen = 0
+
+		// Invalid characters, render as quoted-printable.
 		case b < ' ' || '~' < b:
-			// Invalid character, render quoted-printable into buffer.
-			s := fmt.Sprintf("=%02X", b)
-			n += copy(dest[n:], s)
+			ensureLineLen(2)
```
Comment (dcormier) [inline internal/coding/quotedprint.go:116]: And this one?
```suggestion
			ensureLineLen(3)
```

File: internal/coding/quotedprint.go (hunk 1 above)
Comment (dcormier) [reply internal/coding/quotedprint.go:91]: No, I see. Never mind this one.

File: internal/coding/quotedprint.go (hunk 2 above)
Comment (dcormier) [reply internal/coding/quotedprint.go:116]: And I follow this one, now, too.

**Review by dcormier — APPROVED** [review]
Seems like it should do the trick.

File: internal/coding/quotedprint.go (hunk 1 above)
Comment (jhillyerd) [reply internal/coding/quotedprint.go:92]: I had the same confusion when I wrote it.  :)  I've added clarification to ensureLinLen's comment.

**Pull request merged**